components/scream/src/share/property_checks/field_nan_check.cpp

```diff
@@ -112,10 +130,40 @@ PropertyCheck::CheckResult FieldNaNCheck::check_impl() const {
   }
 
   PropertyCheck::CheckResult check_result;
-  check_result.pass = (num_invalid == 0);
+  check_result.pass = (invalid_idx < 0);
   check_result.msg = "";
   if (not check_result.pass) {
-    check_result.msg = std::string("FieldNaNCheck failed; ") + std::to_string(num_invalid) + " invalid values found\n";
+    auto indices = unflatten_idx(layout.dims(),invalid_idx);
```

Comment:
Is this going to end up reporting ALL of the indices that fail the invalid check? That might end up creating a very large string below?

Reply:
Only one of the columns, yeah. And I think it's the last. Unfortunately, keeping track of all the nan's would be a bit more difficult. Do you think it would be necessary?

Reply:
In terms of NaN, just finding out there is one of them is probably all we need, in my opinion.  Mins/maxes are a different story.  I could see a case where the user would like to know if the list of min/max values is scattered or clustered.  But that is also probably a lot of work.  I would be content to get at least one of the offending columns printed to screen with the location info.

Reply:
Just knowing one location where bad data is occurring would be super helpful. Knowing all the locations would of course be nice, but let's not make perfect be the enemy of the good. We can make another PR some time in the (far) future if we decide that we need all locations. My feeling is that we won't because we just want a general sense of where things are going bad, which is probably all in one area. If it's not, you have bigger problems to deal with.